### ISSUE TITLE
Special fonts are not rendered correctly

### ISSUE THREAD

**Issue by xiaoyangzi** [wojtekmaj/react-pdf#62]
several pdf files is unreadable when rendered:
![unreadable](https://user-images.githubusercontent.com/6845793/30582362-fd91599a-9d55-11e7-8f2b-890554281898.png)
here is the file should be:
![readable](https://user-images.githubusercontent.com/6845793/30582397-1dbd710e-9d56-11e7-84eb-bd477307bcda.png)

**Comment by wojtekmaj** [comment]
Hey @xiaoyangzi,
thanks for reporting. This is an issue with PDF.js, an engine React-PDF uses to render PDFs (Mozilla Firefox is also using it).
Please report it to [mozilla/pdf.js](https://github.com/mozilla/pdf.js) giving them examples of files that render incorrectly.
Thank you in advance,
Wojciech

**Comment by xiaoyangzi** [comment]
This file is rendered correctly by the online viewer Mozilla/pdf.js provide. Maybe it's a version problem that pdf-dist is not sync the pdf.js lateast version  I think.

**Comment by wojtekmaj** [comment]
I'll have a look then. Could you please kindly attach the file you have a problem with?

**Comment by xiaoyangzi** [comment]
[23.pdf](https://github.com/wojtekmaj/react-pdf/files/1316955/23.pdf)

**Comment by wojtekmaj** [comment]
Thank you. Asked for some assistance from Mozilla @ mozilla/pdf.js#8937
Will get back to you if I come up with a solution.

**Comment by wojtekmaj** [comment]
Hey @xiaoyangzi!
The issue is a little complicated. First of all: Some PDF files with exotic characters/fonts may require helper files called cMaps. Mozilla is currently fixing an issue where these cMaps are requested from a null location.

Once they do fix this, I can come and do some enhancements in the code to make it possible to set cMap file location.

Fixing this issue will take some time and will also require you to write some code and configuration once I come up with the solution.

**Comment by KiaFathi** [comment]
+1 request for cMap support. Would a similar workaround to https://github.com/mikecousins/react-pdf-js/issues/16 be possible in this library as well?

**Comment by wojtekmaj** [comment]
Actually yes, even without any changes it should be possible. You would need to do something like:

```js

const pdfjs = require('pdfjs-dist');
pdfjs.PDFJS.cMapUrl = `your_url_to_cmaps_here`;
pdfjs.PDFJS.cMapPacked = true;
```

before initialization of loading PDF and you should be fine!

**Comment by wojtekmaj** [comment]
Hey @xiaoyangzi and @KiaFathi, good news!

Check out [Support for non-latin characters](https://github.com/wojtekmaj/react-pdf#support-for-non-latin-characters) that just landed in v2.3.0!
